gcc/d/dfrontend/expression.c

```diff
@@ -4082,6 +4082,12 @@ Expression *ArrayLiteralExp::semantic(Scope *sc)
         return new ErrorExp();
     }
 
+    if (global.params.noTypeinfo && elements->dim != 0)
+    {
+        error (global.params.noTypeinfo, "Can't use array literal");
```

Comment:
I've just updated old code. Back then visitors were not available and IIRC you can't know in `ArrayLiteralExp::semantic` if it's used for a static array. Anyway, this should probably be rewritten as a visitor.